### PR TITLE
Fix build

### PR DESCRIPTION
**Pull request by ostlerc**
warnings as errors caused build to fail on osx / ubuntu.

### CODE REVIEW

**Comment by tehul** [comment]
Everything looks good. Thanks. I only tested the build using GCC, so I didn't realize Clang had warnings.

File: common/materialContainer/MaterialContainer.h [hunk 1]
```diff
@@ -76,7 +76,7 @@ template <class T> class MaterialContainer
     , contentFreq(MC.contentFreq)
   {
   }
-  MaterialContainer& operator=(const MaterialContainer& mc) { return *this; }
+  MaterialContainer& operator=(const MaterialContainer&) { return *this; }
```
Comment (tehul) [inline common/materialContainer/MaterialContainer.h]: Should we just get rid of this line? It seems like just returning `*this` without assigning the MaterialContainer's fields would be confusing. I guess we could also just assign all the fields if we wanted. Something like,
```c++
  MaterialContainer& operator=(const MaterialContainer& mc)
  {
    name = mc.name;
    contents = mc.contents;
    freq = mc.freq;
    smoothness = mc.smoothness;
    contentFreq = mc.contentFreq;
    return *this;
  }
```

Comment (ostlerc) [inline common/materialContainer/MaterialContainer.h]: I don't understand the need for a copy method on this object. If there is a valid reason, then I believe assigning all the fields is the accurate route.

Comment (tehul) [inline common/materialContainer/MaterialContainer.h]: If by "copy method" you mean an overloaded equals operator, then yeah, I don't see the reason either. No one's using this class yet, so we can reasonably remove it.

Comment (ostlerc) [inline common/materialContainer/MaterialContainer.h]: if by equals operator you mean assignment operator... :)

Comment (tehul) [inline common/materialContainer/MaterialContainer.h]: Ah, those two are completely different. You got me.

Comment (suin3g) [inline common/materialContainer/MaterialContainer.h]: I believe that assignment operator is needed to have a vector of this class.

**Comment by ostlerc** [comment]
We should not be using stack objects in vectors. Either used a shared_ptr or just a regular pointer and you do not need this definition.

File: common/materialContainer/MaterialContainer.h (hunk 1 above)
Comment (ostlerc) [inline common/materialContainer/MaterialContainer.h]: We should not be using stack objects in vectors. Either used a shared_ptr or just a regular pointer and you do not need this definition.

Comment (ksundberg) [inline common/materialContainer/MaterialContainer.h]: The compiler defined assignment operator should be fine.

**Pull request merged**